pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -196,4 +207,42 @@ public InternalConfigurationData getInternalConfigurationData() {
             conf.getZkLedgersRootPath());
     }
 
+    @GET
+    @Path("/health")
+    @ApiOperation(value = "Run a healthcheck against the broker")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Everything is OK"),
+                            @ApiResponse(code = 403, message = "Don't have admin permission"),
+                            @ApiResponse(code = 404, message = "Cluster doesn't exist") })
+    public void healthcheck(@Suspended AsyncResponse asyncResponse) throws Exception {
+        validateSuperUserAccess();
+        pulsar().getExecutor().submit(() -> {
+                String heartbeatNamespace = NamespaceService.getHeartbeatNamespace(
+                        pulsar().getAdvertisedAddress(), pulsar().getConfiguration());
+                String topic = String.format("persistent://%s/healthcheck", heartbeatNamespace);
+                try {
+                    PulsarClient client = pulsar().getClient();
+
+                    try (Producer<String> producer = client.newProducer(Schema.STRING).topic(topic).create();
```

Comment:
Can we rewrite this logic in async way? I know writing sync code is easy, but running blocking operations in pulsar executor is not a good practice.